compaction/compaction_manager.cc

```diff
@@ -566,12 +566,21 @@ future<> compaction_manager::stop_ongoing_compactions(sstring reason) {
     return stop_tasks(std::move(tasks), std::move(reason));
 }
 
-future<> compaction_manager::stop_ongoing_compactions(sstring reason, column_family* cf) {
+future<> compaction_manager::stop_ongoing_compactions(sstring reason, column_family* cf, std::optional<sstables::compaction_type> type_opt) {
     auto ongoing_compactions = get_compactions(cf).size();
-    auto tasks = boost::copy_range<std::vector<lw_shared_ptr<task>>>(_tasks | boost::adaptors::filtered([cf] (auto& task) {
-        return task->compacting_cf == cf;
+    auto tasks = boost::copy_range<std::vector<lw_shared_ptr<task>>>(_tasks | boost::adaptors::filtered([cf, type_opt] (auto& task) {
+        return (!cf || task->compacting_cf == cf) && (!type_opt || task->type == *type_opt);
     }));
-    cmlog.info("Stopping {} tasks for {} ongoing compactions for table {}.{} due to {}", tasks.size(), ongoing_compactions, cf->schema()->ks_name(), cf->schema()->cf_name(), reason);
+    if (cmlog.is_enabled(logging::log_level::info)) {
+        sstring scope = "";
+        if (cf) {
+            scope = format(" for table {}.{}", cf->schema()->ks_name(), cf->schema()->cf_name());
```

Comment:
We should deprecate seastar::format().

Reply:
Although, it does have the nice benefit of generating an sstring (not really needed here)

Reply:
I'll change to fmt::format then.